lightning/src/routing/scoring.rs

```diff
@@ -504,21 +516,36 @@ pub struct ProbabilisticScorerUsingTime<G: Deref<Target = NetworkGraph>, T: Time
 }
 
 /// Parameters for configuring [`ProbabilisticScorer`].
+///
+/// Used to configure a base penalty and a liquidity penalty, the sum of which is the channel
+/// penalty (i.e., the amount in msats willing to be paid to avoid routing through the channel).
 #[derive(Clone, Copy)]
 pub struct ProbabilisticScoringParameters {
-	/// A multiplier used to determine the amount in msats willing to be paid to avoid routing
-	/// through a channel, as per multiplying by the negative `log10` of the channel's success
-	/// probability for a payment.
+	/// A fixed penalty in msats to apply to each channel.
+	///
+	/// Default value: 500 msat
+	pub base_penalty_msat: u64,
+
+	/// The function calculating the cost of routing an amount through a channel.
 	///
-	/// The success probability is determined by the effective channel capacity, the payment amount,
-	/// and knowledge learned from prior successful and unsuccessful payments. The lower bound of
-	/// the success probability is 0.01, effectively limiting the penalty to the range
-	/// `0..=2*liquidity_penalty_multiplier_msat`. The knowledge learned is decayed over time based
-	/// on [`liquidity_offset_half_life`].
+	/// The cost is multiplied by [`liquidity_penalty_multiplier_msat`] to determine the liquidity
+	/// penalty, which is limited to `2 * liquidity_penalty_multiplier_msat`. The cost is based in
+	/// part by the knowledge learned from prior successful and unsuccessful payments. This
+	/// knowledge is decayed over time based on [`liquidity_offset_half_life`].
 	///
-	/// Default value: 10,000 msat
+	/// Default value: [`ProbabilisticScoringCostFunction::NegativeLogSuccessProbability`]
 	///
+	/// [`liquidity_penalty_multiplier_msat`]: Self::liquidity_penalty_multiplier_msat
 	/// [`liquidity_offset_half_life`]: Self::liquidity_offset_half_life
+	pub cost_function: ProbabilisticScoringCostFunction,
+
+	/// A multiplier used in conjunction with [`cost_function`] to determine the liquidity penalty.
+	///
+	/// The liquidity penalty is effectively limited to `2 * liquidity_penalty_multiplier_msat`.
+	///
+	/// Default value: 10,000 msat
```

Comment:
Note this default is probably much too high for the linear case - I don't think we should change it as long as the `cost_function` default is log, but maybe we want to call out in the docs that it should be changed if the cost function is?